tests/stub/authorization/test_temporal_auth_manager.py

```diff
@@ -0,0 +1,319 @@
+                    list(session.run("RETURN 3 AS n"))
+
+                with self.session(driver) as session:
+                    list(session.run("RETURN 4 AS n"))
+
+                self.assertEqual(2, count)
+
+        self._reader.done()
+        self.post_script_assertions(self._reader)
+
+    def test_renewing_on_expiration_error(self):
+        def _test(error_, routing_):
+            count = 0
+
+            def provider():
+                nonlocal count
+                count += 1
+                credentials = "password" if count > 1 else "pass"
+
+                return types.TemporalAuthToken(
+                    types.AuthorizationToken(
+                        scheme="basic",
+                        principal="neo4j",
+                        credentials=credentials
+                    ),
+                    10_000
+                )
+
+            auth_manager = TemporalAuthTokenManager(self._backend, provider)
+
+            expected_call_count = 2 if error_ == "token_expired" else 1
+
+            self.start_server(
+                self._reader,
+                self.script_fn_with_features(f"reader_reauth_{error_}.script")
+            )
+            if routing_:
+                self.start_server(
+                    self._writer,
+                    self.script_fn_with_features(
+                        f"writer_reauth_{error_}.script"
+                    )
+                )
+                self.start_server(self._router, "router_single_reader.script")
+
+            with self.driver(auth_manager, routing=routing_) as driver:
+                if routing_:
+                    with self.session(driver, "w") as session_w:
+                        list(session_w.run("RETURN 1 AS n"))
+
+                with self.session(driver) as session_r1:
+                    with self.session(driver) as session_r2:
+                        with self.session(driver) as session_r3:
+                            # bind connection 1
+                            s1_tx = session_r1.begin_transaction()
+                            list(s1_tx.run("RETURN 1.1 AS n"))
+
+                            self.assertEqual(1, count)
+
+                            # bind connection 2
+                            s2_tx = session_r2.begin_transaction()
+                            list(s2_tx.run("RETURN 2.1 AS n"))
+
+                            # bind connection 3
+                            s3_tx = session_r3.begin_transaction()
+                            list(s3_tx.run("RETURN 3.1 AS n"))
+
+                            s2_tx.commit()
+
+                            self.assertEqual(1, count)
+
+                            with self.assertRaises(types.DriverError) as exc:
+                                # connection 2 fails, gets closed
+                                list(session_r2.run("RETURN 2.2 AS n"))
+                            if error_ == "token_expired":
+                                self.assert_is_token_error(exc.exception)
```

Comment:
This verification might need some update as it should check if this is a retryable instance of the error.